### ISSUE TITLE
Error parsing MySQL56 GTID when SID/UUID repeats

### ISSUE THREAD

**Issue by shlomi-noach** [vitessio/vitess#11890]
Current codebase has a bug parsing MySQL56 GTID values, when the same SID (or UUID) repeats itself. For example, the following GTID set is valid:
```
00021324-1111-1111-1111-111111111111:100-200,00021324-1111-1111-1111-111111111111:300-400
```
canonically, it is equal to:
```
00021324-1111-1111-1111-111111111111:100-200:300-400
```

but the current code erroneously overwrites existing SID entries and so we end up parsing as:
```
00021324-1111-1111-1111-111111111111:300-400
```

**Comment by shlomi-noach** [comment]
Addressed by #11888

**Comment by mattlord** [comment]
At a quick glance, it looks like we probably have the same issue with MariaDB GTIDs: https://github.com/vitessio/vitess/blob/main/go/mysql/mariadb_gtid.go#L79

We no longer support MariaDB outside of MoveTables for imports... but we may still want to fix it there too.

**Comment by shlomi-noach** [comment]
Closed by #11888, but now reopening because we want another PR for MariaDB

**Comment by shlomi-noach** [comment]
@mattlord I'm not very familiar with MariaDB GTID and want to make sure I'm looking at the right thing. could you please provide an example of a GTID set that breaks under existing parsing implementation?
